### PR TITLE
Pv2 399 Fix Acceptance Tests in Dev Environment

### PR DESCRIPTION
**Pull request by samuel-dev-dba**

### CODE REVIEW

File: src/SFA.DAS.Payments.Database/Payments2/Tables/Payment.sql [hunk 1]
```diff
@@ -39,4 +39,16 @@ CREATE INDEX [IX_Payment__ApprenticeshipKey] ON [Payments2].[Payment]
 	LearningAimStandardCode,
 	LearningAimFrameworkCode ,
 	LearningAimPathwayCode 
-)
\ No newline at end of file
+)
+
+GO
+
+CREATE INDEX [IX_Payment__UkprnPeroidSearch] ON [Payments2].[Payment]
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.Database/Payments2/Tables/Payment.sql]: typo - Period not Peroid

File: src/SFA.DAS.Payments.ProviderPayments.Application/Repositories/ProviderPaymentsRepository.cs [hunk 2]
```diff
@@ -52,11 +53,23 @@ public async Task<List<long>> GetMonthEndUkprns(short collectionYear, byte colle
...
             await paymentsDataContext.SaveChangesAsync(cancellationToken);
         }
-        
+
         public async Task SavePayment(PaymentModel paymentData, CancellationToken cancellationToken)
         {
-            await paymentsDataContext.Payment.AddAsync(paymentData, cancellationToken);
-            await paymentsDataContext.SaveChangesAsync(cancellationToken);
+
+            var transactionOptions = new TransactionOptions()
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.ProviderPayments.Application/Repositories/ProviderPaymentsRepository.cs]: include snapshot isolation in sql project

File: src/SFA.DAS.Payments.ProviderPayments.AcceptanceTests/Steps/BindingBootstrapper.cs [hunk 3]
```diff
@@ -30,13 +32,13 @@ public static void SetUpPaymentsDataContext()
         public static void AddRoutingConfig()
         {
             var endpointConfiguration = Container.Resolve<EndpointConfiguration>();
-            endpointConfiguration.Conventions().DefiningEventsAs(type => type.IsEvent<ProviderPaymentEvent>());
+            var conventions = endpointConfiguration.Conventions();
+            conventions.DefiningEventsAs(type => type.IsEvent<IPaymentsEvent>())
```
Comment (abroodie) [inline src/SFA.DAS.Payments.ProviderPayments.AcceptanceTests/Steps/BindingBootstrapper.cs]: The only event this test is concerned with are the events published by the provider payments service.   Needs to be changed back to
endpointConfiguration.Conventions().DefiningEventsAs(type => type.IsEvent<ProviderPaymentEvent>());

File: src/SFA.DAS.Payments.ProviderPayments.AcceptanceTests/Steps/BindingBootstrapper.cs [hunk 4]
```diff
@@ -30,13 +32,13 @@ public static void SetUpPaymentsDataContext()
...
+                .DefiningCommandsAs(t =>  t.IsAssignableTo<ProcessProviderMonthEndCommand>());
             var transportConfig = Container.Resolve<TransportExtensions<AzureServiceBusTransport>>();
             var routing = transportConfig.Routing();
-            routing.RouteToEndpoint(typeof(SfaCoInvestedFundingSourcePaymentEvent), EndpointNames.ProviderPaymentEndPointName);
-            routing.RouteToEndpoint(typeof(EmployerCoInvestedFundingSourcePaymentEvent), EndpointNames.ProviderPaymentEndPointName);
-            routing.RouteToEndpoint(typeof(PerformMonthEndProcessingCommand), EndpointNames.ProviderPaymentEndPointName);
-            routing.RouteToEndpoint(typeof(IlrSubmittedEvent), EndpointNames.ProviderPaymentEndPointName);
+            routing.RouteToEndpoint(typeof(ProcessProviderMonthEndCommand), EndpointNames.ProviderPaymentEndPointName);
```
Comment (abroodie) [inline src/SFA.DAS.Payments.ProviderPayments.AcceptanceTests/Steps/BindingBootstrapper.cs]: Routing config needs to be reverted

File: src/SFA.DAS.Payments.ProviderPayments.AcceptanceTests/Steps/ProviderPaymentsSteps.cs [hunk 5]
```diff
@@ -58,7 +59,7 @@ public async Task WhenTheFundingSourcePaymentsEventAreReceivedAsync()
             var payments = FundingSourcePayments.Select(p => CreateFundingSourcePaymentEvent(p, submissionTime)).ToList();
             foreach (var payment in payments)
             {
-                await MessageSession.Send(payment).ConfigureAwait(false);
+                await MessageSession.Publish(payment).ConfigureAwait(false);
```
Comment (abroodie) [inline src/SFA.DAS.Payments.ProviderPayments.AcceptanceTests/Steps/ProviderPaymentsSteps.cs]: This needs to be a send as we are only testing the provider payments service.

File: src/SFA.DAS.Payments.ProviderPayments.AcceptanceTests/Steps/ProviderResubmitsILRSteps.cs [hunk 6]
```diff
@@ -49,7 +49,7 @@ public async Task GivenTheProviderHasSubmittedAnILRFileWithJobIdWhichHasGenerate
                 CollectionPeriod = new CalendarPeriod(GetYear(CollectionPeriod, CollectionYear).ToString(), CollectionPeriod)
             };
             Console.WriteLine($"Sending the ilr submission event: {ilrSubmissionEvent.ToJson()}");
-            await MessageSession.Send(ilrSubmissionEvent).ConfigureAwait(false);
+            await MessageSession.Publish(ilrSubmissionEvent).ConfigureAwait(false);
```
Comment (abroodie) [inline src/SFA.DAS.Payments.ProviderPayments.AcceptanceTests/Steps/ProviderResubmitsILRSteps.cs]: change this back to send please

File: src/SFA.DAS.Payments.ProviderPayments.AcceptanceTests/Steps/ProviderResubmitsILRSteps.cs [hunk 7]
```diff
@@ -93,14 +93,14 @@ public async Task WhenTheProviderRe_SubmitsAnILRFileWhichTriggersTheFollowingCon
...
```
Comment (abroodie) [inline src/SFA.DAS.Payments.ProviderPayments.AcceptanceTests/Steps/ProviderResubmitsILRSteps.cs]: change this back to send please

File: src/SFA.DAS.Payments.ProviderPayments.AcceptanceTests/Steps/ProviderResubmitsILRSteps.cs [hunk 8]
```diff
@@ -93,14 +93,14 @@ public async Task WhenTheProviderRe_SubmitsAnILRFileWhichTriggersTheFollowingCon
...
 
             ContractType = contractType;
             var fundingSourcePayments = table.CreateSet<FundingSourcePayment>().Select(p => CreateFundingSourcePaymentEvent(p, submissionTime)).ToList();
             foreach (var fundingSourcePaymentEvent in fundingSourcePayments)
             {
                 Console.WriteLine($"Sending funding source event: {fundingSourcePaymentEvent.ToJson()}");
-                await MessageSession.Send(fundingSourcePaymentEvent).ConfigureAwait(false);
+                await MessageSession.Publish(fundingSourcePaymentEvent).ConfigureAwait(false);
```
Comment (abroodie) [inline src/SFA.DAS.Payments.ProviderPayments.AcceptanceTests/Steps/ProviderResubmitsILRSteps.cs]: change this back to send please

File: src/SFA.DAS.Payments.ProviderPayments.AcceptanceTests/Steps/ProviderResubmitsILRSteps.cs [hunk 9]
```diff
@@ -116,7 +116,7 @@ public async Task WhenTheProviderRe_SubmitsAnILRFileWhichTriggersTheFollowingCon
...
```
Comment (abroodie) [inline src/SFA.DAS.Payments.ProviderPayments.AcceptanceTests/Steps/ProviderResubmitsILRSteps.cs]: change this back to send please

File: src/SFA.DAS.Payments.ProviderPayments.AcceptanceTests/Steps/ProviderResubmitsILRSteps.cs [hunk 10]
```diff
@@ -128,7 +128,7 @@ public async Task WhenTheProviderRe_SubmitsAnILRFileWhichTriggersTheFollowingCon
                 CollectionPeriod = new CalendarPeriod(GetYear(CollectionPeriod, CollectionYear).ToString(), CollectionPeriod)
             };
             Console.WriteLine($"Sending the ilr submission event: {ilrSubmissionEvent.ToJson()}");
-            await MessageSession.Send(ilrSubmissionEvent).ConfigureAwait(false);
+            await MessageSession.Publish(ilrSubmissionEvent).ConfigureAwait(false);
```
Comment (abroodie) [inline src/SFA.DAS.Payments.ProviderPayments.AcceptanceTests/Steps/ProviderResubmitsILRSteps.cs]: change this back to send please

File: src/SFA.DAS.Payments.Database/Payments2/Tables/Payment.sql (hunk 1 above)
Comment (samuel-dev-dba) [reply src/SFA.DAS.Payments.Database/Payments2/Tables/Payment.sql]: Done

File: src/SFA.DAS.Payments.ProviderPayments.AcceptanceTests/Steps/BindingBootstrapper.cs (hunk 3 above)
Comment (samuel-dev-dba) [reply src/SFA.DAS.Payments.ProviderPayments.AcceptanceTests/Steps/BindingBootstrapper.cs]: Done

File: src/SFA.DAS.Payments.ProviderPayments.AcceptanceTests/Steps/BindingBootstrapper.cs (hunk 4 above)
Comment (samuel-dev-dba) [reply src/SFA.DAS.Payments.ProviderPayments.AcceptanceTests/Steps/BindingBootstrapper.cs]: Done

File: src/SFA.DAS.Payments.ProviderPayments.AcceptanceTests/Steps/ProviderResubmitsILRSteps.cs (hunk 6 above)
Comment (samuel-dev-dba) [reply src/SFA.DAS.Payments.ProviderPayments.AcceptanceTests/Steps/ProviderResubmitsILRSteps.cs]: Done

File: src/SFA.DAS.Payments.ProviderPayments.AcceptanceTests/Steps/ProviderPaymentsSteps.cs (hunk 5 above)
Comment (samuel-dev-dba) [reply src/SFA.DAS.Payments.ProviderPayments.AcceptanceTests/Steps/ProviderPaymentsSteps.cs]: Done

File: src/SFA.DAS.Payments.ProviderPayments.AcceptanceTests/Steps/ProviderResubmitsILRSteps.cs (hunk 7 above)
Comment (samuel-dev-dba) [reply src/SFA.DAS.Payments.ProviderPayments.AcceptanceTests/Steps/ProviderResubmitsILRSteps.cs]: Done

File: src/SFA.DAS.Payments.ProviderPayments.AcceptanceTests/Steps/ProviderResubmitsILRSteps.cs (hunk 8 above)
Comment (samuel-dev-dba) [reply src/SFA.DAS.Payments.ProviderPayments.AcceptanceTests/Steps/ProviderResubmitsILRSteps.cs]: Done

File: src/SFA.DAS.Payments.ProviderPayments.AcceptanceTests/Steps/ProviderResubmitsILRSteps.cs (hunk 9 above)
Comment (samuel-dev-dba) [reply src/SFA.DAS.Payments.ProviderPayments.AcceptanceTests/Steps/ProviderResubmitsILRSteps.cs]: Done

File: src/SFA.DAS.Payments.ProviderPayments.AcceptanceTests/Steps/ProviderResubmitsILRSteps.cs (hunk 10 above)
Comment (samuel-dev-dba) [reply src/SFA.DAS.Payments.ProviderPayments.AcceptanceTests/Steps/ProviderResubmitsILRSteps.cs]: Done

File: src/SFA.DAS.Payments.Database/SFA.DAS.Payments.Database.sqlproj [hunk 11]
```diff
@@ -27,6 +27,7 @@
     <IncludeSchemaNameInFileName>True</IncludeSchemaNameInFileName>
     <Recovery>SIMPLE</Recovery>
     <DacApplicationName>SFA.DAS.Payments</DacApplicationName>
+    <AllowSnapshotIsolation>True</AllowSnapshotIsolation>
```
Comment (kuzinator) [inline src/SFA.DAS.Payments.Database/SFA.DAS.Payments.Database.sqlproj]: You have enabled snapshot isolation here but I can't see it used anywhere.
I needs either setting READ_COMMITTED_SNAPSHOT to ON or set explicitly in every session.

Comment (samuel-dev-dba) [reply src/SFA.DAS.Payments.Database/SFA.DAS.Payments.Database.sqlproj]: AllowSnapshotIsolation now set to false

**Pull request merged**